### PR TITLE
Add cancel method to the debounce function

### PR DESCRIPTION
**Pull request by EvandroLG**
I refactored the debounce function a little bit, adding a method to cancel the debounced function.
If you - @angus-c - are okay with this implementation, afterwards I will add the same method in the throttle function as well.

### CODE REVIEW

File: test/function-debounce/index.js [hunk 1]
```diff
@@ -18,6 +18,23 @@ test('waits for n ms then runs once', function(t) {
...
+  fn();
+  fn();
+  fn.cancel();
+
+  setTimeout(function() {
+    t.equal(callCounter, 0);
+  }, 300);
+});
+
 test('when callFirst is true, runs once, waits for n ms then runs again', function(
```
Comment (angus-c) [inline test/function-debounce/index.js:38]: Can you add another test (positioned after "when CallFirst is true" test)

```
test('cancel debounced function when callFirst is true', function(t) {
  t.plan(1);

  var callCounter = 0;
  var fn = debounce(function() {
    callCounter++;
  }, 200, true);

  fn();
  fn();
  fn.cancel();

  setTimeout(function() {
    t.equal(callCounter, 1);
  }, 300);
});
```

**Review by angus-c — CHANGES_REQUESTED** [review]
Thank you!

Please also update `index.tests.ts` so it starts like this

```
// OK
debounce(() => {});
debounce(() => {}, 50);
debounce(() => {}, 200, true);
debounce(() => {}, 1000, false);

debounce(() => {}).cancel();
debounce(() => {}, 50).cancel();
debounce(() => {}, 200, true).cancel();
debounce(() => {}, 1000, false).cancel();

// not OK
// @ts-expect-error
debounce();
// @ts-expect-error
debounce().cancel();

[//etc...]
```

**Comment by EvandroLG** [comment]
Done!
Btw, how are you running the `index.test.ts` file? `ts-node`?

**Comment by angus-c** [comment]
> Done!
> Btw, how are you running the `index.test.ts` file? `ts-node`?

Thanks! I’m AFK right now but will land/version/publish when I get a chance

I do `tsc <test name>` and try to remember to delete generated js before pushing 🤣(I guess I could add a got ignore for that pattern)

But also vscode linter highlights any ts errors  as I add them

**Comment by EvandroLG** [comment]
Maybe, it would be nice to add a script to run those tests as we have for lint and unit tests. WDYT?

**Comment by angus-c** [comment]
Yeah agree. I only looked into it very briefly. Not sure what the exit codes etc for that script are.

**Comment by angus-c** [comment]
Published as function-debounce@1.4.0

**Pull request merged**